Deploy dalite server

jbzdak opened the pull request.

antoviaque commented on host_vars/dalite.harvardx.harvard.edu/public.yml:
```diff
@@ -0,0 +1,47 @@
+---
+
+MYSQL_DALITE_HOST: "149.202.183.147"
```
This setting should be moved to the private vars? The (public) default should be non-functional imho, to avoid people reusing this from hammering an existing server if they don't touch the config.

jbzdak replied:
Addressed in 75bc82946df447e3a9700bfd85d153c6f41de359,

antoviaque commented on group_vars/all/public.yml:
```diff
@@ -0,0 +1,11 @@
+tarsnap_cron_hour: "*"
+tarsnap_tarsnapper_conf: "{{ playbook_dir }}/opencraft-roles/roles/tarsnap/files/conf/tarsnapper.{{ inventory_hostname }}.conf"
+
+OPS_EMAIL: 'jacek@opencraft.com'
```
This should also be set publicly to a non-functional email (ops@example.com), and overriden in the private vars.

jbzdak replied:
Addressed in 75bc82946df447e3a9700bfd85d153c6f41de359,

antoviaque commented on group_vars/all/public.yml:
```diff
@@ -0,0 +1,11 @@
+COMMON_SERVER_ETCKEEPER_COMMIT_EMAIL: '{{ OPS_EMAIL }}'
+security_autoupdate_mail_to: '{{ OPS_EMAIL }}'
+
+FORWARD_MAIL_SMTP_FROM: "ops@opencraft.com"
```
Same thing

jbzdak replied:
Addressed in 75bc82946df447e3a9700bfd85d153c6f41de359,

antoviaque commented on host_vars/dalite.harvardx.harvard.edu/public.yml:
```diff
@@ -0,0 +1,47 @@
+---
```
Also this variables file is pretty generic, shared by all hosts - can we use the values here for all hosts, and simply override things specific to some hosts in the private config?

jbzdak replied:
Abstract: Not really, we can specify global overrides, and we can't override variables for a given host. 

Not really. The problem is the variable file precedence in Ansible. Ansible people also dislike overriding variables, and say that: "If you have to figure out variable from which file is used you are doing something wrong" (I'm not saying that this right, but that's the general attitude). 

Anyways, ``host_vars`` and ``group_vars`` can be in two places: in the directory where the executed playbook is (in our case it's deploy repository --- or this repository) and in the directory where ansible.cfg (in our case secrets repository). However for ansible ``host_vars`` from ```deploy`` repository takes precedence. 

The approach I use is to define in public.yml things like: 

``TARSNAP_KEY = "{{ MYSQL_TARSNAP_KEY }}"``

and then in private yml I can set tarsnap key, for only a single instance setting ``MYSQL_TARSNAP_KEY``.

antoviaque replied:
But I mean, why should the `host_vars` be public, and why should it contain the default values? It doesn't make sense, especially since the `hosts` file is private. This should only contain host-specific configuration overrides, not the default configuration to share with every host. 

Can you move the default configuration to the role defaults (http://docs.ansible.com/ansible/playbooks_roles.html#role-default-variables) and move `host_vars` and its remaining host-specific variables to the private repo?

jbzdak replied:
@antoviaque This was added here after our [discussion here](https://github.com/open-craft/deployment-deploy-services/pull/1#discussion_r60221179). Most of these variables are also role defaults, but I needed to move them to top-level scope so other roles (swift backup, sanity checker) has access to them. Previously I just hardcoded a couple of paths, which seemed to be lesser evil. 

When we have second dalite instance deployed from this repository, we might as well create a ``group_vars/dalite`` and it'll work. 

Of course I could cram all variables to private vars... but then: 

1. Value of this repository would be much smaller for another people 
2. Deploying new instances of dalite will be much less fun.

jbzdak replied:
@antoviaque This has been addressed in 773dbd5156111ac3ee98503128449bc64202521e. No variables that we don't want to share between all instances are left in this file.

antoviaque commented:
@jbzdak Executing the playbook fails with:

```
antoviaque@antoviaque-laptop:~/prog/edx/devops/deployment-secrets-services$ ./venv/bin/ansible-galaxy install -r deploy/requirements.yml -f && ./venv/bin/ansible-playbook deploy/deploy-all.yml -u admin --extra-vars private-extra-vars.yaml
- extracting geerlingguy.security to deploy/roles/geerlingguy.security
- geerlingguy.security was installed successfully
- extracting geerlingguy.nginx to deploy/roles/geerlingguy.nginx
- geerlingguy.nginx was installed successfully
- extracting kamaln7.swapfile to deploy/roles/kamaln7.swapfile
- kamaln7.swapfile was installed successfully
- extracting pmbauer.tarsnap to deploy/roles/pmbauer.tarsnap
- pmbauer.tarsnap was installed successfully
- extracting backup-to-tarsnap to deploy/roles/backup-to-tarsnap
- backup-to-tarsnap was installed successfully
[DEPRECATION WARNING]: The comma separated role spec format, use the yaml/explicit format instead.. This feature will be removed in a future release. Deprecation warnings can be disabled by setting 
deprecation_warnings=False in ansible.cfg.
- dependency pmbauer.tarsnap already pending installation.
- extracting common-server to deploy/roles/common-server
- common-server was installed successfully
- dependency geerlingguy.security already pending installation.
- dependency kamaln7.swapfile already pending installation.
- extracting configure-apt to deploy/roles/configure-apt
- configure-apt was installed successfully
- extracting sanity-checker to deploy/roles/sanity-checker
- sanity-checker was installed successfully
- extracting forward-server-mail to deploy/roles/forward-server-mail
- forward-server-mail was installed successfully
- extracting backup-swift-container to deploy/roles/backup-swift-container
- backup-swift-container was installed successfully
- dependency pmbauer.tarsnap already pending installation.
- extracting dalite to deploy/roles/dalite
- dalite was installed successfully
- dependency geerlingguy.nginx already pending installation.
ERROR! the role 'ansible-backup-to-tarsnap' was not found in /home/antoviaque/prog/edx/devops/deployment-secrets-services/deploy/roles:/home/antoviaque/prog/edx/devops/deployment-secrets-services/deploy:deploy/roles:/home/antoviaque/prog/edx/devops/deployment-secrets-services/deploy/roles

The error appears to have been in '/home/antoviaque/prog/edx/devops/deployment-secrets-services/deploy/roles/backup-swift-container/meta/main.yml': line 3, column 5, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

  - pmbauer.tarsnap
  - name: ansible-backup-to-tarsnap
    ^ here
```

jbzdak commented:
> ERROR! the role 'ansible-backup-to-tarsnap' was not found 

@antoviaque Sorry for that issue. Long story short I had stale ``roles`` and everything worked for me. I have now checked that I can perform deployment from scratch for dalite.

antoviaque commented on README.md:
```diff
@@ -1 +1,102 @@
+------------------------------
+
+1. Create a root volume that has 40GB size. It should be based on Ubuntu 14.04 image. 
+   We suggest you use vanilla image from [ubuntu page](https://cloud-images.ubuntu.com/). 
```
@jbzdak Where are the instructions to do that?

jbzdak replied:
Addressed in: 7efa934d1d98bcce3948a6ece1dfbc45512bc22e

omarkhan replied:
@jbzdak I'm not sure that the changes in 7efa934 explain how to use the vanilla ubuntu image. Is the ubuntu image in the image list the vanilla image or OVH's image? I think you have to upload the image first using the openstack cli:

    glance image-create --disk-format=qcow2 --container-format=bare --file <img> --name <name> --progress

@antoviaque can you confirm?

jbzdak replied:
@omarkhan Uploading is covered in a separate document, which since there are more than one deployment repository is here: https://github.com/open-craft/public-doc/pull/1 .

antoviaque commented on README.md:
```diff
@@ -1 +1,102 @@
+
+1. Create a root volume that has 40GB size. It should be based on Ubuntu 14.04 image. 
+   We suggest you use vanilla image from [ubuntu page](https://cloud-images.ubuntu.com/). 
+2. Create `vps-ssd-3` instance from this volume: (`Boot Source` -> `Volume` -> `Your root volume`). 
```
I'm not sure this does what you think it does - If you create a volume based on the stock ubuntu image, and then boot from that volume, it will just run the installation from the volume, not install the system on it?

omarkhan commented on README.md:
```diff
@@ -1 +1,103 @@
+    1. Generate key for swift container backup 
+    2. Generate key for dalite logrotate backup   
+2. Generate tarsnap read write keys from the master key, see [tarsnap-keymngmt](http://www.tarsnap.com/man-tarsnap-keymgmt.1.html),    
+3. Go to dead man's snitch at https://deadmanssnitch.com/. And generate three snitches, save them under:
```
What interval should be used for these snitches?

jbzdak replied:
I'll amend the readme, to unblock you: 

1. DALITE_LOG_TARSNAP_SNITCH - daily
2. SANITY_CHECK_SNITCH 15min
3. BACKUP_SWIFT_SNITCH hourly

antoviaque commented:
:+1: once the remaining comments are addressed

omarkhan commented:
@jbzdak tested and working 👍

Commit:
Initial version of deploy services repository.

Squashed commit of the following:

commit 116e90fdc5cbe3af61da4013a5f4f80f458677f0
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Tue May 3 18:18:56 2016 +0200

    Use hashes instead of branches for repeatable builds

commit 69990c3f2797c324ca179a81df63cd47ae78e4e1
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Mon May 2 23:56:04 2016 +0200

    Use branch for deployment.

commit 0b6aeecafeb9e75d8514a159eaa4cf48b3d9d971
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Mon May 2 20:44:52 2016 +0200

    Renamed logs download directory

commit 7a14f28bd7cca539b2ba55371f912c2860164b16
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Fri Apr 29 21:26:37 2016 +0200

    Fixed sending emails from initial provisioning run

commit d6ca515ca245516c199a8e317160b969774f5d72
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Fri Apr 29 20:44:47 2016 +0200

    Upading hostname as the first part of provisioning.

commit 7efa934d1d98bcce3948a6ece1dfbc45512bc22e
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Fri Apr 29 15:46:08 2016 +0200

    Updated readme

commit 518d71662abb7226c7a0a9dc9e125b6774554b3d
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Fri Apr 29 15:12:06 2016 +0200

    Updated readme.

commit 088d78c45436599efd407f77147c9d8f93b43b1f
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Fri Apr 29 15:09:06 2016 +0200

    Updated readme.

commit 6ed036141402b1519fbfdd51a9571392ac261839
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Fri Apr 29 13:00:03 2016 +0200

    Updated readme

commit 773dbd5156111ac3ee98503128449bc64202521e
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Fri Apr 29 12:58:45 2016 +0200

    Moved variables shared by all dalites to group_vars.

commit 60f7e5ecc53659f43d347b78661eca3e39634817
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Thu Apr 28 10:47:55 2016 +0200

    Removed DALITE_MEDIA_CONTAINER from public.yml and moved to private.

    It's not really sensitive,but not a sensible default for others
    also.

commit 12a009ac149dbfcc102b984b1a68dcf9dfe129a0
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Wed Apr 27 19:07:04 2016 +0200

    Added license file

commit 7e29a5a07a3e973a9869806d98fc793030791ed6
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Wed Apr 27 17:50:34 2016 +0200

    Updated readme to fit what is done right now.

commit 75bc82946df447e3a9700bfd85d153c6f41de359
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Wed Apr 27 16:08:29 2016 +0200

    Address PR-review comments

commit 218b2f2f54142abc362733121870da5738050740
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Tue Apr 26 18:02:21 2016 +0200

    Updated mysql host

commit c00ad6c1810981a2150bf640f1d553a75fa6c050
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Tue Apr 26 17:44:44 2016 +0200

    Bumped dalite version

commit 6c19777ab9ad4bf751efc3e10dc7cd37d6f8c9d5
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Mon Apr 25 18:35:13 2016 +0200

    Dropped cryptography requirement

commit dbba8552a7d2dafab1c96a092514b370907a696a
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Mon Apr 25 15:45:55 2016 +0200

    Fix typos

commit 2977de3ab4d818dd16ef47e391cde8107a191d53
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Mon Apr 18 17:40:51 2016 +0200

    Deploy dalite server

Commit:
PR review comments

jbzdak commented:
Review comments have been addressed in a563e9e2adb3cfcabc8a12d3b0b3899cba357414
